### ISSUE TITLE
--rd flag makes stylelint not to report any violations at all

### ISSUE THREAD

**Issue by sobolevn** [stylelint/stylelint#4100]
> Clearly describe the bug

Reproduction:

```css
/* index.css */
.empty{}
```

Running `npx stylelint index.css`:

```
1:8  ✖  Unexpected empty block   block-no-empty
```

Then, let's add an unused ignore:

```css
/* index.css */
/* stylelint-disable-next-line property-no-unknown */
.empty{}
```

And run `npx stylelint --rd index.css`. No output will be received. It just passes.

Expected output:
1. `block-no-empty` violation
2. Unused `property-no-unknown` comment violation

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "block-no-empty": true
  }
}
```

> Which version of stylelint are you using?

latest: 9.10.1

**Comment by sobolevn** [comment]
Original issue: https://github.com/wemake-services/wemake-vue-template/issues/608

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

> block-no-empty violation

When `--report-needless-disables` stylelint reports only needless disables, and no other violations. We have an [open issue](https://github.com/stylelint/stylelint/issues/4076) to change that.

> Unused property-no-unknown comment violation

I can't reproduce it. With `stylelint@10.1.0` I see the report:

```
$ npx stylelint index.css --report-needless-disables

index.css
unused rule: property-no-unknown, start line: 2, end line: 2
```

**Comment by hudochenkov** [comment]
Closing because of no response from the author.

**Comment by sobolevn** [comment]
I would say that this issue: https://github.com/stylelint/stylelint/issues/4076 tracks the feature I am interested in.
